noir-projects/aztec-nr/aztec/src/oracle/header.nr

```diff
@@ -30,22 +34,59 @@ pub fn get_header_at(block_number: u32, context: PrivateContext) -> Header {
             last_archive_block_number >= block_number, "Last archive block number is smaller than the block number we want to prove against"
         );
 
-        // 3) Get the header of a given block from oracle
-        let header = get_header_at_internal(block_number);
-
-        // 4) Compute the block hash from the block header
-        let block_hash = header.hash();
+        // 3) Get the header hint of a given block from an oracle
+        let header_hint = get_header_at_internal(block_number);
 
-        // 5) Get the membership witness of the block in the archive
-        let witness = get_archive_membership_witness(last_archive_block_number, block_hash);
-
-        // 6) Check that the block is in the archive (i.e. the witness is valid)
-        assert(
-            context.historical_header.last_archive.root
-            == root_from_sibling_path(block_hash, witness.index, witness.path), "Proving membership of a block in archive failed"
+        // 4) We make sure that the header hint we received from the oracle exists in the state tree and is the actual header
+        // at the desired block number
+        constrain_get_header_at_internal(
+            header_hint,
+            block_number,
+            last_archive_block_number,
+            historical_header.last_archive.root
         );
 
-        // 7) Return the block header
-        header
+        // 8) Return the block header
```

Comment:
8? Also we can maybe get rid of some of these numbering comments as they're sort of superfluous as the function is simplified